expression/integration_test.go

```diff
@@ -1797,8 +1797,8 @@ func (s *testIntegrationSuite) TestTimeBuiltin(c *C) {
 		{"\"2011-01-01 00:00:00\"", "10.10", "DAY", "2011-01-11 00:00:00", "2010-12-22 00:00:00"},
 		{"\"2011-01-01 00:00:00\"", "10.10", "HOUR", "2011-01-01 10:00:00", "2010-12-31 14:00:00"},
 		{"\"2011-01-01 00:00:00\"", "10.10", "MINUTE", "2011-01-01 00:10:00", "2010-12-31 23:50:00"},
-		{"\"2011-01-01 00:00:00\"", "10.10", "SECOND", "2011-01-01 00:00:10", "2010-12-31 23:59:50"},
-		{"\"2011-01-01 00:00:00\"", "10.10", "MICROSECOND", "2011-01-01 00:00:00.000010", "2010-12-31 23:59:59.999990"},
+		{"\"2011-01-01 00:00:00\"", "10.10", "SECOND", "2011-01-01 00:00:10.100000", "2010-12-31 23:59:49.900000"},
+		{"\"2011-01-01 00:00:00\"", "10.10", "MICROSECOND", "2011-01-01 00:00:00.000010", "2010-12-31 23:59:59.999989"},
```

Comment:
What happens with this one?

```
mysql> select subdate('2011-01-01 00:00:00', interval 10.10 microsecond);
+------------------------------------------------------------+
| subdate('2011-01-01 00:00:00', interval 10.10 microsecond) |
+------------------------------------------------------------+
| 2010-12-31 23:59:59.999990                                 |
+------------------------------------------------------------+
1 row in set (0.00 sec)
```

Reply:
This is a really problem. In MYSQL, it's rounding. But it round down here. I'll check it.